src/MockBot.UnitTests/Controllers/ChatControllerTests.cs

```diff
@@ -83,44 +80,63 @@ public void ShouldCreateAndReturnChat()
         [InlineData("Some text")]
         public async Task ShouldAddMessageToChatAsync(string payload)
         {
+            // Arrange
             _sut.ControllerContext = new ControllerContext()
             {
                 HttpContext = GetContext(payload)
             };
 
-            var message = new ChatMessage(payload);
-            var chat = new Chat();
+            var chat = _chatService.CreateChat();
             var currentDateTime = new DateTime();
-            _ = _mockChatService.Setup(mock => mock.AddMessage(chat.Id, payload)).Returns(message);
 
+            // Act
             var result = await _sut.PostMessageAsync(chat.Id).ConfigureAwait(false);
 
+            // Assert
             var createdResult = Assert.IsType<CreatedResult>(result);
             var resultMessage = Assert.IsType<ChatMessage>(createdResult.Value);
             Assert.Equal($"/chats/{chat.Id}/messages", createdResult.Location);
             Assert.NotEmpty(resultMessage.Id.ToString());
             Assert.True(currentDateTime < resultMessage.CreatedAt);
-            _mockChatService.Verify(mock => mock.AddDmrRequest(message));
-            _mockDmrService.Verify(mock => mock.Enqueue(It.IsAny<DmrRequest>()));
         }
 
         [Theory]
         [InlineData("Some text")]
         public async Task ShouldFailToAddMessageToChatWhenNoChatWithGivenIdAsync(string payload)
         {
+            // Arrange
             _sut.ControllerContext = new ControllerContext()
             {
                 HttpContext = GetContext(payload)
             };
-            var chat = new Chat();
-            _ = _mockChatService.Setup(mock => mock.AddMessage(chat.Id, payload)).Throws(new ArgumentOutOfRangeException(chat.Id.ToString()));
+            var chatId = Guid.NewGuid();
 
+            // Act
+            var result = await _sut.PostMessageAsync(chatId).ConfigureAwait(false);
 
-            var result = await _sut.PostMessageAsync(chat.Id).ConfigureAwait(false);
-
+            // Assert
             _ = Assert.IsType<NotFoundObjectResult>(result);
         }
 
+        [Fact]
+        public async Task ShouldReturnBadReqestIfBodyEmpty()
+        {
+            // Arrange
+            _sut.ControllerContext = new ControllerContext()
+            {
+                HttpContext = GetContext(string.Empty)
+            };
+            var chatId = Guid.NewGuid();
+
+            // Act
+            var result = await _sut.PostMessageAsync(chatId).ConfigureAwait(false);
+
+            // Assert
+            _ = Assert.IsType<BadRequestObjectResult>(result);
+            var resultBadRequest = result as BadRequestObjectResult;
+            Assert.Equal("Post must have a body", resultBadRequest.Value);
```

Comment:
Would it be useful to put our error messages into a constant class shared both by the unit tests and api namespaces, would make it re-usable but I think more importantly, imagine if we update the error message on the controller, this test will fail, so having a single source of truth I guess

Reply:
Good spot, I'll do that

Reply:
Have made the suggested changes